libcni/api.go

```diff
@@ -242,18 +311,76 @@ func (c *CNIConfig) getCachedResult(netName, cniVersion string, rt *RuntimeConf)
 	return result, err
 }
 
+func (c *CNIConfig) getCachedResult(netName, cniVersion string, rt *RuntimeConf) (types.Result, error) {
+	fname := c.getResultCacheFilePath(netName, rt)
+	fdata, err := ioutil.ReadFile(fname)
+	if err != nil {
+		// Ignore read errors; the cached result may not exist on-disk
+		return nil, nil
+	}
+
+	cachedInfo := cachedInfo{}
+	if err := json.Unmarshal(fdata, &cachedInfo); err != nil {
```

Comment:
I think judging condition here could be `err != nil || cachedInfo.Kind != CNICacheV1` to spare four lines of the following `if`.

Reply:
What is returned to caller is different for each if (or I don't understand your comment.)

Reply:
I mean that we can compress
```
	cachedInfo := cachedInfo{}
	if err := json.Unmarshal(fdata, &cachedInfo); err != nil {
		return c.getLegacyCachedResult(netName, cniVersion, rt)
	}

	if cachedInfo.Kind != CNICacheV1 {
		return c.getLegacyCachedResult(netName, cniVersion, rt)
	}
```
into 
```
	cachedInfo := cachedInfo{}
	if err := json.Unmarshal(fdata, &cachedInfo); err != nil || cachedInfo.Kind != CNICacheV1{
		return c.getLegacyCachedResult(netName, cniVersion, rt)
	}
```